### ISSUE TITLE
Invalid or Incomplete POST Params

### ISSUE THREAD

**Issue by mzagaja** [MAPC/myschoolcommute2#130]
@allthesignals while I tested your front end fix to make sure it worked to allow keyboard selection, I failed to test it on the rails end. When digging into batch mode I found the following problem:

```
Invalid or incomplete POST params
Started POST "/survey_responses" for 127.0.0.1 at 2017-08-08 15:36:40 -0400

Rack::QueryParser::ParameterTypeError (expected Hash (got String) for param `grade_1'):
```

This occurs when submitting a survey via the regular or batch interface. The issue goes away when reverting #128, so going to assign as a missed bug in #128 to @allthesignals. As a hint [StackOverflow suggests it is an issue with how you organize your params](https://stackoverflow.com/questions/9259210/expected-hash-got-array-for-param-samples).

**Comment by allthesignals** [comment]
@mzagaja do you need my help on this?

**Comment by mzagaja** [comment]
I think so. Although I have also just reproduced the bug after the revert, so this is apparently intermittent. At a minimum let me know if it reproduces on your machine. I'm trying to figure out how to trigger it reliably, but might not be the result of your commit, or alternatively might just have caused increased frequency of the bug. The params don't look wrong in web inspector but I'm now trying to figure out how to hook into the debugger on the rails side to get better information.

**Comment by mzagaja** [comment]
Yeah just as an update it seems to reliably reproduce after your commit, but happens once in a while without it. I'm happy to try and tag team this thing because it doesn't seem fun at all.

**Comment by allthesignals** [comment]
@mzagaja the React application just generates DOM... in this case it's creating [form elements](https://www.w3schools.com/html/html_form_input_types.asp). Maybe they aren't parsing because something is wrong with the input field names? Rails uses those to parse nested hashes.

When you use the browser inspector, you can see the name attributes of fields. For that one, I see this:
`survey_response[grade_1]-search`

It should be 
`survey_response[grade_1]`. 

I'll make a PR.

**Comment by allthesignals** [comment]
Alright, I've identified the issue but my PR doesn't address it. I'll take another look tonight!

**Comment by mzagaja** [comment]
@allthesignals Thanks, I was just testing it and was a little confused why it didn't seem different. You are a gentleman 👱, scholar 🎓, and true friend of the community 🏡. I'll keep an eye out for the update tonight.

**Comment by mzagaja** [comment]
Also FYI I pulled the params fix into develop at https://github.com/MAPC/myschoolcommute2/commit/5279e0014440aa8b2508475302f1a0fc3163a364

**Comment by mzagaja** [comment]
Marking as fixed in #132, but either way, is not reproducing currently.
